baseplate/experiments/__init__.py

```diff
@@ -90,6 +90,18 @@ def _get_experiment(self, name):
             self._experiment_cache[name] = experiment
         return self._experiment_cache[name]
 
+    def _get_all_experiment_names(self):
+        config = self._config_watcher.get_data()
+        exp_names = []
+        if config:
+            for name, cfg in config.items():
+                exp_names.append(name)
+
+        return exp_names
+
+    def is_valid_experiment(self, name):
```

Comment:
🔕 Rather than providing this, it might make more sense to have `variant` raise an exception if the experiment name does not map to a known experiment.  I'm not 100% sure about this as it has some downsides:

1. It means that code will randomly break when experiments get removed, in some ways this is _good_ because it forces you to clean up your code but it will likely break things.
2. It also means that we have to update the r2 code to account for this.
3. It means that you _have_ to put up your experiment config before you deploy your code that uses the experiment, this could be a pro or a con depending on how you look at it.

Reply:
> It means that code will randomly break when experiments get removed... 
This the primary reason I didn't go that route. Given that experiments can be disabled/removed from an external source (like an experiment config/setup tool), and that by people who are not the implementing engineers, I think it's just too risky to throw exceptions. I don't like the idea that someone could remove an experiment via the configuration system and break systems that haven't properly caught that exception.

> in some ways this is good because it forces you to clean up your code
I don't love breaking systems as a forcing function for a process. Cleaning up old code is a process problem, and probably shouldn't be solved with the threat of broken systems. 

> It means that you have to put up your experiment config before you deploy your code that uses the experiment
This is also not ideal. Experiments are deliberately run for a fixed time period, and having to coordinate the setup and deployment of experimental code to get close to your experiment window increases complication. I think it's better to allow the code to go out, then flip the experiment on when ready.

Reply:
if this is meant to be a public function, it'll need a docstring so it shows up in docs.

Reply:
added